app/grandchallenge/archives/forms.py

```diff
@@ -68,16 +70,33 @@ def __init__(self, *args, **kwargs):
             .distinct()
         )
         if self.instance:
-            interface_slugs = (
-                self.instance.items.exclude(values__isnull=True)
-                .values_list("values__interface__slug", flat=True)
+            interfaces = (
+                ComponentInterface.objects.filter(
+                    componentinterfacevalue__in=self.instance.items.exclude(
+                        values__isnull=True
+                    ).values_list("values", flat=True)
+                )
                 .order_by()
                 .distinct()
```

Comment:
Why not just do `self.instance.items.exclude(values__isnull=True).distinct()` to get the interfaces like you do for algorithms, for example? 

And is the `order_by` necessary? And if so, would it make sense to do that also in the other places? 

In general it looks like you're doing the same thing in multiple places. If so, I think this should be a mixin - reduces the logic to one place, makes it easier to maintain.

Reply:
And the following is probably more efficient:

```suggestion
                interface_slugs = self.instance.items.exclude(values__isnul=True).values_list("values__slug", flat=True)
                ComponentInterface.objects.filter(
                    slugs__in=interface_slugs
                )
```

Reply:
the `order_by` is not necessary.
Agree with the more efficient query.
Done.